### PR TITLE
Add astroquery

### PR DESCRIPTION
**Pull request by stsci-hack**
This PR adds all the functionality necessary to use astroquery in the HLAPipeline for getting specific datasets.  The stand-alone functionality was then incorporated into the test classes to transparently handle getting data from astroquery as well as Artifactory or the local cache.  

A new test was also added to specifically test/demonstrate this new functionality.

### CODE REVIEW

File: hlapipeline/utils/astroquery_utils.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    path : string
+        Directory to use for writing out downloaded files.  If `None` (default),
+        the current working directory will be used.
+
+    """
+    local_files = []
+    obsTable = Observations.query_criteria(obs_id=obsid)
+    # Catch the case where no files are found for download
+    if len(obsTable) < 0:
```
Comment (mdlpstsci) [inline hlapipeline/utils/astroquery_utils.py]: It seems like this should be `len(obsTable) <= 0`?

File: tests/base_classes.py [hunk 2]
```diff
@@ -88,14 +90,18 @@ def get_data(self, *args, **kwargs):
         `artifactory_helpers/get_bigdata()`.
         This will then return the full path to the local copy of the file.
```
Comment (mdlpstsci) [inline tests/base_classes.py]: I would add a comment here indicating the use of astroquery from an archive versus obtaining the data from Artifactory or the local store.

Comment (stsci-hack) [reply tests/base_classes.py]: Addressed with e77079e.

**Comment by stsci-hack** [comment]
Logic error identified in review corrected in 2ad8084

**Pull request merged**